services/wallet/reader.go

```diff
@@ -32,29 +36,34 @@ type Reader struct {
 	cancel       context.CancelFunc
 }
 
+type TokenMarketValues struct {
+	MarketCap       float64 `json:"marketCap"`
+	HighDay         float64 `json:"highDay"`
+	LowDay          float64 `json:"lowDay"`
+	ChangePctHour   float64 `json:"changePctHour"`
+	ChangePctDay    float64 `json:"changePctDay"`
+	ChangePct24hour float64 `json:"changePct24hour"`
+	Change24hour    float64 `json:"change24hour"`
+	Price           float64 `json:"price"`
+}
+
 type ChainBalance struct {
 	Balance *big.Float     `json:"balance"`
 	Address common.Address `json:"address"`
 	ChainID uint64         `json:"chainId"`
 }
 
 type Token struct {
-	Name             string                  `json:"name"`
-	Symbol           string                  `json:"symbol"`
-	Color            string                  `json:"color"`
-	Decimals         uint                    `json:"decimals"`
-	BalancesPerChain map[uint64]ChainBalance `json:"balancesPerChain"`
-	Description      string                  `json:"description"`
-	AssetWebsiteURL  string                  `json:"assetWebsiteUrl"`
-	BuiltOn          string                  `json:"builtOn"`
-	MarketCap        string                  `json:"marketCap"`
-	HighDay          string                  `json:"highDay"`
-	LowDay           string                  `json:"lowDay"`
-	ChangePctHour    string                  `json:"changePctHour"`
-	ChangePctDay     string                  `json:"changePctDay"`
-	ChangePct24hour  string                  `json:"changePct24hour"`
-	Change24hour     string                  `json:"change24hour"`
-	CurrencyPrice    float64                 `json:"currencyPrice"`
+	Name                    string                       `json:"name"`
+	Symbol                  string                       `json:"symbol"`
+	Color                   string                       `json:"color"`
+	Decimals                uint                         `json:"decimals"`
+	BalancesPerChain        map[uint64]ChainBalance      `json:"balancesPerChain"`
+	Description             string                       `json:"description"`
+	AssetWebsiteURL         string                       `json:"assetWebsiteUrl"`
+	BuiltOn                 string                       `json:"builtOn"`
+	MarketValuesPerCurrency map[string]TokenMarketValues `json:"marketValuesPerCurrency"`
+	Peg                     string                       `json:"peg"`
```

Comment:
This name sounds like a fit for a boolean value. Won't PegSymbol make it more straightforward?

Reply:
Did some extra reading to see what names are used in the literature.
The "boolean" state is commonly referred to as "pegged". (USDT is pegged to the US Dollar)
"Peg" seems to commonly refer to a value + a currency. (USDT has a peg of 1 USD).
I don't think there is one, but I guess there could be a peg with a value different than 1 (pegged to 100 USD?), so I'm guessing your suggestion makes sense and we should call this "PegSymbol". Will do the change.

Reply:
Fixed